python/llm/src/bigdl/llm/cli/llm-cli.sh

```diff
@@ -0,0 +1,474 @@
+  echo "  --repeat_last_n N                 last n tokens to consider for penalize (default: 64, 0 = disabled, -1 = ctx_size)"
+  echo "  --repeat_penalty N                penalize repeat sequence of tokens (default: 1.1, 1.0 = disabled)"
+  echo "  --presence_penalty N              repeat alpha presence penalty (default: 0.0, 0.0 = disabled)"
+  echo "  --frequency_penalty N             repeat alpha frequency penalty (default: 0.0, 0.0 = disabled)"
+  echo "  --mirostat N                      use Mirostat sampling."
+  echo "                                    Top K, Nucleus, Tail Free and Locally Typical samplers are ignored if used."
+  echo "                                    (default: 0, 0 = disabled, 1 = Mirostat, 2 = Mirostat 2.0)"
+  echo "  --mirostat_lr N                   Mirostat learning rate, parameter eta (default: 0.1)"
+  echo "  --mirostat_ent N                  Mirostat target entropy, parameter tau (default: 5.0)"
+  echo "  -l, --logit-bias TOKEN_ID(+/-)BIAS modifies the likelihood of token appearing in the completion,"
+  echo "                                    i.e. '--logit-bias 15043+1' to increase likelihood of token ' Hello',"
+  echo "                                    or '--logit-bias 15043-1' to decrease likelihood of token ' Hello'"
+  echo "  -c, --ctx_size N                  size of the prompt context (default: 512)"
+  echo "  --ignore-eos                      ignore end of stream token and continue generating (implies --logit-bias 2-inf)"
+  echo "  --no-penalize-nl                  do not penalize newline token"
+  echo "  --memory_f32                      use f32 instead of f16 for memory key+value"
+  echo "  --temp N                          temperature (default: 0.8)"
+  echo "  --n_parts N                       number of model parts (default: -1 = determine from dimensions)"
+  echo "  -b, --batch_size N                batch size for prompt processing (default: 512)"
+  echo "  --perplexity                      compute perplexity over the prompt"
+  echo "  --keep N                          number of tokens to keep from the initial prompt (default: 0, -1 = all)"
+  echo "  --mlock                           force system to keep model in RAM rather than swapping or compressing"
+  echo "  --no-mmap                         do not memory-map model (slower load but may reduce pageouts if not using mlock)"
+  echo "  --mtest                           compute maximum memory usage"
+  echo "  --verbose-prompt                  print prompt before generation"
+  echo "  --lora FNAME                      apply LoRA adapter (implies --no-mmap)"
+  echo "  --lora-base FNAME                 optional model to use as a base for the layers modified by the LoRA adapter"
+  echo "  -m, --model FNAME                 model path (default: ./model.bin)"
+  echo "  -x, --model_family {llama,bloom,gptneox} family name of model"
+}
+
+function llama {
+  echo "Unsupported params:"
```

Comment:
Why are these parameters not unsupported for llama?

Reply:
This code is based on an older version of llama.cpp. After updating, all parameters will be passed directly to main.